fansi/shared/src/main/scala/fansi/Fansi.scala

```diff
@@ -679,10 +700,31 @@ object Back extends Category(offset = 12, width = 9){
     for(x <- 0 to 256)
     yield makeAttr(s"\u001b[48;5;${x}m", 17 + x)(s"Back.Full($x)")
 
-  val all = Vector(
+  def True(hex: Long) = {
+    val r = hex >> 16
+    val g = (hex & 0x00FF00) >> 8
+    val b = hex & 0x0000FF
+    makeAttr(s"\u001b[48;2;${r};${g};${b}m", 174 + hex)(s"Back.True($r,$g,$b)")
+  }
+
+  def True(r: Int, g: Int, b: Int) = {
+    val i = r << 16 | g << 8 | b
+    makeAttr(s"\u001b[48;2;${r};${g};${b}m", 174 + i)(s"Back.True($r,$g,$b)")
+  }
+
+  val all: Vector[Attr] = Vector(
     Reset, Black, Red, Green, Yellow, Blue, Magenta, Cyan, LightGray, DarkGray,
     LightRed, LightGreen, LightYellow, LightBlue, LightMagenta, LightCyan, White
   ) ++ Full
+
+  override def lookupAttr(applyState : Long) : Attr = {
+    val index = applyState >> offset
+    if(index < 274 ) {
+      lookupAttrTable(index.toInt) //I think it's safe if i > 0
+    } else {
+      True(index)
+    }
+  }
```

Comment:
Could we abstract these custom implementations of `True`, `lookupAttr` into a single trait rather than than duplicating the logic?